flowey/flowey_core/src/pipeline.rs

```diff
@@ -664,27 +676,50 @@ impl Pipeline {
         self
     }
 
+    /// Format a parameter name based on it's kind (stable / unstable).
+    ///
+    /// `name` is the name of the parameter.
+    ///
+    /// `kind` is the type of parameter and if it should be treated as a stable
+    /// external API to callers of the pipeline.
+    fn new_parameter_name(&mut self, name: impl AsRef<str>, kind: ParameterKind) -> String {
```

Comment:
the function looks good, but there's no reason to hang it off the `impl Pipeline`. You're not actually using `self` here or anything. This could just be a free-standing `fn new_parameter_name(name: impl ...)` outside the `impl Pipeline` block, just in the same module.

Reply:
also, you don't strictly _need_ to document this method, given that it's entirely private... but more docs ain't ever a bad thing :)